sqlmesh/core/model/definition.py

```diff
@@ -530,8 +541,12 @@ def columns_to_types_or_raise(self) -> t.Dict[str, exp.DataType]:
     @property
     def annotated(self) -> bool:
         """Checks if all column projection types of this model are known."""
-        columns_to_types = self.columns_to_types
-        if columns_to_types is None:
+        if self.columns_to_types is None:
+            return False
+        columns_to_types = {
+            k: v for k, v in self.columns_to_types.items() if k not in self.managed_columns
```

Comment:
Why does it matter whether it's a managed column or not?

Reply:
This method is telling us if all column projections are known (either explicitly by user or inferred from the expression). We obviously know the types of the columns we manage. Therefore when doing this check we just want to consider the user defined columns in order to determine if we have figured out the annotations.